addons/web/static/src/core/debug/debug_context.js

```diff
@@ -60,17 +60,27 @@ class DebugContext {
     }
 }
 
+let currentDebugContext;
 const debugContextSymbol = Symbol("debugContext");
 export function useOwnDebugContext({ categories = [] } = {}) {
-    useSubEnv({ [debugContextSymbol]: new DebugContext(useEnv(), categories) });
+    const oldDebugContext = currentDebugContext;
+    currentDebugContext = new DebugContext(useEnv(), categories);
+    useSubEnv({ [debugContextSymbol]: currentDebugContext });
+    useEffect(
+        () => {
+            return () => {
+                currentDebugContext = oldDebugContext;
+            };
+        },
+        () => []
+    );
 }
 
 export function useCurrentDebugContext() {
-    const debugContext = useEnv()[debugContextSymbol];
-    if (!debugContext) {
+    if (!currentDebugContext) {
         throw new Error("There is no debug context available in the current environment.");
     }
-    return debugContext;
+    return currentDebugContext;
```

Comment:
I think the hook that returns the "current" context should be a separate function. With this implementation, if a view dialog is open and some part of the webclient under it renders a new component, that component it will activate debug categories *inside* the view dialog even though it's not in it. Maybe this hook can be renamed to "useEnvDebugContext" or something though. Also, the current name is misleading: it looks like a hook but it isn't (it uses no builtin hooks and can be used from a service which can't use hooks), the original hook used the component environment.